nixos/modules/services/networking/gns3-server.nix

```diff
@@ -0,0 +1,267 @@
+      };
+
+      port = lib.mkOption {
+        type = lib.types.port;
+        default = 3080;
+        description = lib.mdDoc ''Port on which GNS3 Server will listen on.'';
+      };
+
+      log = {
+        file = lib.mkOption {
+          type = lib.types.nullOr lib.types.path;
+          default = "/var/log/gns3/server.log";
+          description = lib.mdDoc ''Path of the file GNS3 Server should log to.'';
+        };
+
+        debug = lib.mkOption {
+          default = false;
+          type = lib.types.bool;
+          description = lib.mdDoc ''Whether to enable debug logging.'';
+        };
+      };
+
+      tls = {
+        enable = lib.mkOption {
+          type = lib.types.bool;
+          default = false;
+          description = lib.mdDoc ''Whether to enable TLS encryption.'';
+        };
+
+        certFile = lib.mkOption {
+          type = lib.types.nullOr lib.types.path;
+          default = null;
+          example = "/secrets/gns3.crt";
+          description = lib.mdDoc ''
+            Path to the TLS certificate file. This certificate will
+            be offered to, and may be verified by, clients.
+          '';
+        };
+
+        keyFile = lib.mkOption {
+          type = lib.types.nullOr lib.types.path;
+          default = null;
+          example = "/secrets/gns3.key";
+          description = lib.mdDoc "Private key file for the certificate.";
+        };
+      };
+
+      dynamips = {
+        enable = lib.mkEnableOption (lib.mdDoc ''Whether to enable Dynamips support.'');
+        package = lib.mkPackageOptionMD pkgs "dynamips" { };
+      };
+
+      ubridge = {
+        enable = lib.mkEnableOption (lib.mdDoc ''Whether to enable uBridge support.'');
+        package = lib.mkPackageOptionMD pkgs "ubridge" { };
+      };
+
+      vpcs = {
+        enable = lib.mkEnableOption (lib.mdDoc ''Whether to enable VPCS support.'');
+        package = lib.mkPackageOptionMD pkgs "vpcs" { };
+      };
+    };
+  };
+
+  config = lib.mkIf cfg.enable {
+    environment.systemPackages = [ cfg.package ]
+      ++ lib.optional cfg.dynamips.enable cfg.dynamips.package
+      ++ lib.optional cfg.ubridge.enable cfg.ubridge.package
+      ++ lib.optional cfg.vpcs.enable cfg.vpcs.package;
+
+    users.groups.ubridge = lib.mkIf cfg.ubridge.enable { };
+
+    security.wrappers.ubridge = lib.mkIf cfg.ubridge.enable {
+      capabilities = "cap_net_raw,cap_net_admin=eip";
+      group = "ubridge";
+      owner = "root";
+      permissions = "u=rwx,g=rx,o=r";
+      source = "${cfg.ubridge.package}/bin/ubridge";
+    };
+
+    users.users.gns3 = {
+      name = "gns3";
+      group = "gns3";
+      description = "GNS3 user";
+      isSystemUser = true;
+      home = "/var/lib/gns3";
+      extraGroups = lib.optional flags.enableDocker "docker"
+        ++ lib.optional flags.enableLibvirtd "libvirtd"
+        ++ lib.optional cfg.ubridge.enable "ubridge";
+    };
+
+    users.groups.gns3 = { };
```

Comment:
I did not use DynamicUser because I was unsure it would be "safe to use" with a stateful service like GNS3 server. But after using it for a few hours, it seems to be working normally.

So as suggested, I enabled DynamicUser on the service, and removed the gns3 user/group to use the SupplementaryGroups property instead.

Many thanks for the review! @h7x4